src/test/java/de/hpi/bpt/argos/api/EntityMappingEndpointTest.java

```diff
@@ -0,0 +1,355 @@
+		EventEntityMapping testMapping = createMapping();
+		String oldStatus = testMapping.getTargetStatus();
+		int entityMappings = getMappingsCount();
+
+		RestRequest request = RestRequestFactoryImpl.getInstance()
+				.createPutRequest(ARGOS_REST_HOST, getEditEntityMappingUri(testMapping.getId()));
+
+		JsonObject newMapping = createNewMappingJson();
+		newMapping.addProperty("EventTypeId", 0);
+
+		request.setContent(serializer.toJson(newMapping));
+
+		assertEquals(HttpStatusCodes.BAD_REQUEST, request.getResponseCode());
+
+		assertEquals(entityMappings, getMappingsCount());
+
+		assertMapping(PersistenceAdapterImpl.getInstance().getEventEntityMapping(testMapping.getId()), oldStatus);
+	}
+
+	@Test
+	public void testEditEntityMapping_InvalidEntityAttributeTypeId_BadRequest() {
+		EventEntityMapping testMapping = createMapping();
+		String oldStatus = testMapping.getTargetStatus();
+		int entityMappings = getMappingsCount();
+
+		RestRequest request = RestRequestFactoryImpl.getInstance()
+				.createPutRequest(ARGOS_REST_HOST, getEditEntityMappingUri(testMapping.getId()));
+
+		JsonObject newMapping = createNewMappingJson();
+		newMapping.get("EventEntityMappingConditions")
+				.getAsJsonArray().get(0).getAsJsonObject()
+				.addProperty("EntityTypeAttributeId", 0);
+
+		request.setContent(serializer.toJson(newMapping));
+
+		assertEquals(HttpStatusCodes.BAD_REQUEST, request.getResponseCode());
+
+		assertEquals(entityMappings, getMappingsCount());
+
+		assertMapping(PersistenceAdapterImpl.getInstance().getEventEntityMapping(testMapping.getId()), oldStatus);
+	}
+
+	@Test
+	public void testEditEntityMapping_InvalidEventAttributeTypeId_BadRequest() {
+		EventEntityMapping testMapping = createMapping();
+		String oldStatus = testMapping.getTargetStatus();
+		int entityMappings = getMappingsCount();
+
+		RestRequest request = RestRequestFactoryImpl.getInstance()
+				.createPutRequest(ARGOS_REST_HOST, getEditEntityMappingUri(testMapping.getId()));
+
+		JsonObject newMapping = createNewMappingJson();
+		newMapping.get("EventEntityMappingConditions")
+				.getAsJsonArray().get(0).getAsJsonObject()
+				.addProperty("EventTypeAttributeId", 0);
+
+		request.setContent(serializer.toJson(newMapping));
+
+		assertEquals(HttpStatusCodes.BAD_REQUEST, request.getResponseCode());
+
+		assertEquals(entityMappings, getMappingsCount());
+
+		assertMapping(PersistenceAdapterImpl.getInstance().getEventEntityMapping(testMapping.getId()), oldStatus);
+	}
+
+	private EventEntityMapping createMapping() {
+		EventEntityMapping testMapping = ArgosTestUtil.createEventEntityMapping(testEventType, testEntityType, "", true);
+		List<MappingCondition> testMappingConditions = ArgosTestUtil
+				.createMappingConditions(testMapping, testEventTypeAttributes, testEntityTypeAttributes, true);
+
+		return testMapping;
+	}
+
+	private int getMappingsCount(EventType owner) {
+		return PersistenceAdapterImpl.getInstance().getEventEntityMappingsForEventType(owner.getId()).size();
```

Comment:
This method would only be used for testing. This will cost a bit of performance when testing (since more than needed is fetched from the database) but there won't be an unused method in the productive system.